README.md

```diff
@@ -0,0 +1,367 @@
+- To possess such characteristics, the liquidity module should provide most convenient ways for external users to come in and use the services provided by the Cosmos Hub.
+- The liquidity module should not anticipate specific assets, such as Atom, into the process of user-flow in a forced manner. It is repeatedly proved that unnatural anticipation of native coin at unavoidable parts of process resulting in poor user attraction.
+
+## Key features
+
+![new-amm-model](doc/img/new-amm-model.png)
+
+**Combination of traditional orderbook-based model and new AMM model**
+
+- Although new AMM model has multiple advantages over orderbook-based model,
+combination of both models will create more enriched utilities for wider potential users.
+- This module redefines the concept of a “swap order” in AMM as a “limit order with short lifetime”
+in an orderbook-based exchange. Then, two concepts from two different models can be
+combined as one united model so that the function can provide both ways to participate
+into the trading and liquidity providing activities.
+- Although our first version of the liquidity module will not provide limit order option, but
+the base structure of the codebase is already anticipating such feature expansion in the
+near future.
+- Advantages of combined model
+    - More freedom on how to provide liquidity : Limit orders
+    - Combination of pool liquidity and limit order liquidity provides users more enriched trading environment
+
+For detailed Mechanism, you can find on our recent [Paper](doc/LiquidityModuleLightPaper_EN.pdf)
+
+## Installation
+
+### Requirements
+
+| Requirement | Notes            |
+| ----------- | ---------------- |
+| Go version  | Go1.15 or higher |
+| Cosmos-SDK  | v0.42.4          |
+
+### Get Liquidity Module source code 
+
+```bash 
+$ git clone https://github.com/tendermint/liquidity.git
+$ cd liquidity
+$ go mod tidy
+```
+
+### Build
+
+```bash 
+# You can find the liquidityd binary inside build directory
+$ make build 
+```
+
+### Install
+```bash 
+$ make install 
+```
+
+## Usage of CLI Commands
+
+### Transactions
+
+`$ liquidityd tx liquidity --help`     
+
+```bash
+Liquidity transaction subcommands
+
+Usage:
+  liquidityd tx liquidity [flags]
+  liquidityd tx liquidity [command]
+
+Available Commands:
+  create-pool Create liquidity pool and deposit coins
+  deposit     Deposit coins to a liquidity pool
+  swap        Swap offer coin with demand coin from the liquidity pool with the given order price
+  withdraw    Withdraw pool coin from the specified liquidity pool
+```
+
+### Queries
+
+`$ liquidityd query liquidity --help`
+
+```bash
+Querying commands for the liquidity module
+
+Usage:
+  liquidityd query liquidity [flags]
+  liquidityd query liquidity [command]
+
+Available Commands:
+  batch       Query details of a liquidity pool batch
```

Comment:
If this module is not targeting to be one of the Cosmos SDK modules, then I think it would be fine; however, to follow convention with other Cosmos SDK modules, we should stay with Query.

Examples: 
- https://github.com/cosmos/cosmos-sdk/blob/master/x/staking/client/cli/query.go#L21
- https://github.com/cosmos/cosmos-sdk/blob/master/x/gov/client/cli/query.go#L50
- https://github.com/cosmos/cosmos-sdk/blob/master/x/distribution/client/cli/query.go#L21